BUG: backtrace function detected but not present (aarch64)

### Describe the issue:

The backtrace function is detected, but compilation fails with:

```
-c ../numpy/_core/src/multiarray/temp_elide.c
../numpy/_core/src/multiarray/temp_elide.c:150:13: error: call to undeclared function 'backtrace'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  150 |     nptrs = backtrace(buffer, NPY_MAX_STACKSIZE);
      |             ^
```

<del>
In the meson build file we have:

```
optional_misc_funcs = [
  'backtrace',
  'madvise',
]
foreach func: optional_misc_funcs
  if cc.has_function(func,
      include_directories: inc_curdir,
      prefix: '#include "feature_detection_misc.h"'
    )
    cdata.set10('HAVE_' + func.to_upper(), true)
  endif
endforeach
```

So, first I thought that the problem lies with `feature_detection_misc.h` being included, but that's not it. The test passes without it too.

</del>

Compiler is clang 17.

### Reproduce the code example:

```python
N/A
```


### Error message:

```shell
N/A
```


### Python and NumPy Versions:

Python 3.11, meson 1.3.1, meson-python 0.15.0

### Runtime Environment:

_No response_

### Context for the issue:

_No response_

--- a/numpy/_core/src/multiarray/temp_elide.c
+++ b/numpy/_core/src/multiarray/temp_elide.c
@@ -59,6 +59,9 @@
  */
 
 #if defined HAVE_BACKTRACE && defined HAVE_DLFCN_H && ! defined PYPY_VERSION
+
+#include <feature_detection_misc.h>
+
 /* 1 prints elided operations, 2 prints stacktraces */
 #define NPY_ELIDE_DEBUG 0
 #define NPY_MAX_STACKSIZE 10

